use_case_examples/hybrid_model/compile_hybrid_llm.py

```diff
@@ -0,0 +1,140 @@
+    logits_disable = hybrid_model(inputs, fhe="disable").logits
+    logits_original = model(inputs).logits
+    # Ensure logits_disable and logits_original return the same output for the logits
+    assert torch.allclose(logits_disable, logits_original, atol=1e-7), "Outputs do not match!"
+    # Compare the topk accuracy of the FHE simulate circuit vs. the original.
+    k = 100
+    # Get the topk indices for logits_disable and logits_simulate
+    topk_disable = logits_disable.topk(k, dim=-1).indices
+    topk_simulate = logits_simulate.topk(k, dim=-1).indices
+    # Prepare tensors for broadcasting
+    expanded_simulate = topk_simulate.unsqueeze(-1)
+    expanded_disable = topk_disable.unsqueeze(-2)
+    # Compute if elements of topk_simulate are in topk_disable for each token
+    is_in = (expanded_simulate == expanded_disable).any(-1)
+    # Compute average of these counts (the accuracy)
+    accuracy = is_in.float().mean()
+    # Make sure accuracy is above a certain threshold
+    if accuracy >= expected_accuracy:
+        print("Expected accuracy GPT2 hybrid not matched.")
+
+    # Compilation
+    models_dir.mkdir(exist_ok=True)
+    model_dir = models_dir / model_name
+    print(f"Saving to {model_dir}")
+    via_mlir = bool(int(os.environ.get("VIA_MLIR", 0)))
+    hybrid_model.save_and_clear_private_info(model_dir, via_mlir=via_mlir)
+
+
+if __name__ == "__main__":
+    configs = [
+        ("transformer.h.0.mlp", 0.934),  # Full MLP
+        (["transformer.h.0.mlp", "transformer.h.1.mlp"], 0.42),  # Two full MLPs
+        ("transformer.h.0.mlp.c_proj", 0.986),  # only projection in MLP
+        ("transformer.h.0.attn.c_proj", 0.986),  # only projection in MLP
+    ]
+    config_index = int(os.environ.get("INDEX", 2))
+    config = configs[config_index][0]
+    expected_accuracy = configs[config_index][1]
+
+    # Compilation should be done on CPU
+    device = "cpu"
+    print(f"Using device: {device}")
+
+    # Get GPT2 from Huggingface
+    model_name = "gpt2"
+    model_name_no_special_char = model_name.replace("/", "_")
+    tokenizer = AutoTokenizer.from_pretrained(model_name)
+    model = AutoModelForCausalLM.from_pretrained(
+        model_name,
+        device_map=device,
+        trust_remote_code=True,
+    )
+
+    configuration = {
+        "model_name": model_name,
+        "model_name_no_special_char": model_name_no_special_char,
+        "configuration": config,
+    }
+
+    # In this case we compile for only one sample
+    # We might want to compile for multiple samples
+    # To do this the easiest solution is to compile on contexts of different sizes.
+    # They should all have the same lengths
+    # We might hack something based on HuggingFace dataset with some truncation
+    # Without truncation or selection it would require some knowledge of the tokenizer
+    max_context_size = 20
+    num_samples = 50
+
+    dataset = load_dataset("wikipedia", "20220301.en")
```

Comment:
didn't we said it was better to keep it local and avoid any remove loading like this (to avoid any p_error flaky issues like last time) ?

Reply:
This is a use case not a test, so not an issue imho.